src/payment/bolt12.rs

```diff
@@ -95,6 +96,9 @@ impl Bolt12Payment {
 					preimage: None,
 					secret: None,
 					offer_id: offer.id(),
+					payer_note: payer_note.map(UntrustedString),
+					payer_id: None,
```

Comment:
> Mh, this is a bit weird as internally to LDK we very well know the `payer_id` at this point. Maybe we should consider returning the `InvoiceRequest` from `pay_for_offer`.
> 
> For consistency it would also be good to actually set the payer note based on the invoice request field rather than what the user gave us.

Should I open an issue on rust-lightning before making the change, or should I just make the change and open a PR directly?